lib/exercises.js

```diff
@@ -1,17 +1,100 @@
+
+// function isAnagram(a, b) {
+//   if (a.length != b.length) return false;
+
+//   a = a.split("").sort();
+//   b = b.split("").sort();
+
+//   for (let i = 0; i < a.length; ++i) {
+//     if (a[i] !== b[i]) return false;
+//   }
+//   return true;
+// }
+
+// Time: O(q), where q is the string length
+// Space: O(n), where n is the number of unique characters in either of the strings
+function isAnagram(a, b) {
+  if (a.length != b.length) return false;
+
+  let aObj = {};
+  let bObj = {};
+
+  for (let i = 0; i < a.length; i++) {
+    if (aObj[a[i]]) {
+      aObj[a[i]] += 1;
+    } else {
+      aObj[a[i]] = 1;
+    }
+
+    if (bObj[b[i]]) {
+      bObj[b[i]] += 1;
+    } else {
+      bObj[b[i]] = 1;
+    }
+  }
+
+  const aKeys = Object.keys(aObj);
+  for (let key of aKeys) {
+    if (aObj[key] !== bObj[key]) return false;
+  }
+
+  return true;
+}
+
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(p * q), where p is the number of strings in the input array, and q is the string length
+// Space Complexity: O(n), where n is the number of strings in the input array
 function grouped_anagrams(strings) {
```

Comment:
Consider sorting the letters and using those sorted letters as keys.  It might be easier to code, although the same time-complexity.